unit_tests/src/filesystem/test_filesystem.cpp

```diff
@@ -180,6 +264,81 @@ TEST_CASE("filesystem")
         REQUIRE(fs::client::remove(conn, col, fs::remove_options::no_trash));
     }
 
+    SECTION("fetch data object mtime")
+    {
+        using namespace std::chrono_literals;
+
+        const fs::path p = sandbox / "data_object";
+
+        {
+            irods::experimental::client_connection conn;
+            default_transport tp{conn};
+            odstream{tp, p} << "hello world!";
+        }
+
+        const auto first_replica_mtime = fs::client::last_write_time(conn, p);
```

Comment:
Would it be worth adding a test for the appropriate filesystem error for attempting to modify the last_write_time of a data object? I removed the test when the function was moved to the replica library, but maybe good for completeness.

Reply:
sure.  will add that.

Reply:
added test.